### PR TITLE
Prepare repo for publishing to npm

### PR DESCRIPTION
**Pull request by reefdog**
## Description
This PR prepares the repo for publishing to npm.

- Ensures we build the app before publishing
- Ensures we build the app before testing (which becomes redundant with the above, but is still safest)
- Reverts package version numbers to 0.0.0 so our first `lerna publish` process will work as expected
- Makes sure we publish both the original ES6 and transpiled files to npm
- Decorates each child package's `package.json` with repository and licensing info
- Adds a README for each child package (for completeness, and for rendering on npmjs.com)

Once this PR is merged into master, it should be safe to run `lerna publish` and away we go!

## Due Diligence Checklist
- [x] Tests
- [x] Documentation
- [x] Consolidated commits
- [x] Relevant labels assigned

## Steps to Test
1. `yarn test`

## Related Issues
Affects #13 but doesn't resolve it, since that will be a manual step
Relates to #11

### CODE REVIEW

**Comment by slifty** [comment]
I think you swapped #13 and #11

**Comment by slifty** [comment]
jk ok real review incoming soon I swear

File: packages/classes/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# Changelog for @tvkitchen/base-classes
```
Comment (slifty) [inline packages/classes/CHANGELOG.md:1]: Could / should we use a template inspired by [this one](https://github.com/benmosher/eslint-plugin-import/blob/master/CHANGELOG.md) for our changelogs?

Basically, have the common header text that links to semver and that keepachangelog site (which we should check out!)

Comment (reefdog) [reply packages/classes/CHANGELOG.md:1]: Yup, that's great, and we should! I'm going to modify these with the Unreleased header, and add the actual work done thus far in each.

And [Keepachangelog](https://keepachangelog.com/en/1.0.0/) is fantastic!

**Comment by reefdog** [comment]
You already approved this, but I just pushed some changes to the CHANGELOGs (o no where do I log that??), can you give it another look?

**Comment by slifty** [comment]
Oh, could we update our repository field in the package.jsons to use this kind of structure:

```
"repository": {
  "type" : "git",
  "url" : "https://github.com/facebook/react.git",
  "directory": "packages/react-dom"
}
``` 

I noticed that babel just made a change along the same lines: https://github.com/babel/babel/commit/242d9e0b20af3ec3f05c929eb0cf5a57cc576e39

**Pull request merged**